commit 3be0ad8b0a6a89d2a6ed04124f25912039ee2a30
Author: Martin Zimmermann <info@posativ.org>
Date:   Sun Apr 3 13:42:34 2016 +0200

    remove drv_libxml2 from preferred xml parsers, if we get a TypeError, #10

diff --git a/canto_next/remote.py b/canto_next/remote.py
index bbfe9d6..e5c71fb 100644
--- a/canto_next/remote.py
+++ b/canto_next/remote.py
@@ -164,6 +164,10 @@ class CantoRemote(PluginHandler, CantoClient):
                 'Canto/0.9.0 + http://codezen.org/canto-ng' }
         try:
             content = feedparser.parse(URL, request_headers = extra_headers)
+        except TypeError:
+            if 'drv_libxml2' in feedparser.PREFERRED_XML_PARSERS:
+                feedparser.PREFERRED_XML_PARSERS.remove('drv_libxml2')
+                content = feedparser.parse(URL, request_headers = extra_headers)
         except Exception as e:
             print("ERROR: Couldn't determine name: %s" % e)
             return None
